<p>The use of parentheses, even those not required to enforce a desired order of operations, can clarify the intent behind a piece of code. But
redundant pairs of parentheses could be misleading, and should be removed. </p>
<h2>Noncompliant Code Example</h2>
<pre>
If a AndAlso ((x + y &gt; 0)) Then ' Noncompliant
    ' ...
End If

Return ((x + 1))  ' Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
If a AndAlso x + y &gt; 0 Then
    ' ...
End If

Return (x + 1)
</pre>

