<h2>Why is this an issue?</h2>
<p>This rule is meant to be used as a way to track code which is marked as being deprecated. Deprecated code should eventually be removed.</p>
<h3>Noncompliant code example</h3>
<pre>
&lt;Obsolete&gt; ' Noncompliant
Sub Procedure()
End Sub
</pre>

