<p><code>End</code> statements exit the control flow of the program in an unstructured way. This statement stops code execution immediately without
executing <code>Dispose</code> or <code>Finalize</code> methods, or executing <code>Finally</code> blocks. Therefore, it should be avoided.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Module Module1
    Sub Print(ByVal str As String)
       Try
            ...
            End       ' Noncompliant
        Finally
            ' do something important here
            ...
        End Try
    End Sub
End Module
</pre>

