<p>Throwing an exception from within a finally block will mask any exception which was previously thrown in the <code>try</code> or <code>catch</code>
block, and the maskedâ€™s exception message and stack trace will be lost.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Try
    ' Some work which end up throwing an exception
    Throw New ArgumentException()
Finally
    ' Clean up
    Throw New InvalidOperationException()       ' Noncompliant; will mask the ArgumentException
End Try
</pre>
<h2>Compliant Solution</h2>
<pre>
Try
    ' Some work which end up throwing an exception
    Throw New ArgumentException()
Finally
    ' Clean up
End Try
</pre>

