<p>This rule applies whenever an <code>If</code> statement is followed by one or more <code>ElseIf</code> statements; the final <code>ElseIf</code>
should be followed by an <code>Else</code> statement.</p>
<p>The requirement for a final <code>Else</code> statement is defensive programming.</p>
<p>The <code>Else</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is
consistent with the requirement to have a final <code>Case Else</code> clause in a <code>Select Case</code> statement.</p>
<h2>Noncompliant Code Example</h2>
<pre>
If x = 0 Then
    DoSomething()
ElseIf x = 1 Then
    DoSomethingElse()
End If
</pre>
<h2>Compliant Solution</h2>
<pre>
If x = 0 Then
    DoSomething()
ElseIf x = 1 Then
    DoSomethingElse()
Else
    Throw New ArgumentException("...")
End If
</pre>
<h2>Exceptions</h2>
<p>None</p>
<h2>See</h2>
<ul>
  <li> <a href="https://wiki.sei.cmu.edu/confluence/x/RtYxBQ">CERT, MSC01-C.</a> - Strive for logical completeness </li>
  <li> <a href="https://wiki.sei.cmu.edu/confluence/x/jzZGBQ">CERT, MSC57-J.</a> - Strive for logical completeness </li>
</ul>

