<h2>Why is this an issue?</h2>
<p><code>switch</code> statements are useful when there are many different cases depending on the value of the same expression.</p>
<p>For just one or two cases, however, the code will be more readable with <code>if</code> statements.</p>
<h3>Noncompliant code example</h3>
<pre>
Select Case variable
    Case 0
        doSomething()
    Case Else
        doSomethingElse()
End Select
</pre>
<h3>Compliant solution</h3>
<pre>
If variable = 0 Then
    doSomething()
Else
    doSomethingElse()
End If
</pre>

