<p>There is no reason to re-assign a variable to itself. Either this statement is redundant and should be removed, or the re-assignment is a mistake
and some other value or variable was intended for the assignment instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Public Sub SetName(name As String)
  name = name
End Sub
</pre>
<h2>Compliant Solution</h2>
<pre>
Public Sub SetName(name As String)
  Me.name = name
End Sub
</pre>

