<h2>Why is this an issue?</h2>
<p>There is no reason to re-assign a variable to itself. Either this statement is redundant and should be removed, or the re-assignment is a mistake
and some other value or variable was intended for the assignment instead.</p>
<h3>Noncompliant code example</h3>
<pre>
Public Sub SetName(name As String)
  name = name
End Sub
</pre>
<h3>Compliant solution</h3>
<pre>
Public Sub SetName(name As String)
  Me.name = name
End Sub
</pre>

