<p>The use of non-short-circuit logic in a boolean context is likely a mistake - one that could cause serious program errors as conditions are
evaluated under the wrong circumstances.</p>
<h2>Noncompliant Code Example</h2>
<pre>
If GetTrue() Or GetFalse() Then ' Noncompliant; both sides evaluated
End If
</pre>
<h2>Compliant Solution</h2>
<pre>
If GetTrue() OrElse GetFalse() Then ' true short-circuit logic
End If
</pre>

