<h2>Why is this an issue?</h2>
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all even names match a provided regular expression.</p>
<p>The default configuration is the one recommended by Microsoft:</p>
<ul>
  <li>Pascal casing, starting with an upper case character, e.g. BackColor</li>
  <li>Short abbreviations of 2 letters can be capitalized, e.g. GetID</li>
  <li>Longer abbreviations need to be lower cased, e.g. GetHtml</li>
</ul>
<h3>Noncompliant code example</h3>
<p>With the default regular expression <code>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</code>:</p>
<pre>
Class Foo
    Event fooEvent() ' Noncompliant
End Class
</pre>
<h3>Compliant solution</h3>
<pre>
Class Foo
    Event FooEvent() ' Compliant
End Class
</pre>

