<p>Calling the <code>Not</code> operator twice does nothing: the second invocation undoes the first. Either this is a bug, if the operator was
actually meant to be called once, or misleading if done on purpose.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Dim b As Boolean = False
Dim c As Boolean = Not Not b 'Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
Dim b As Boolean = False
Dim c As Boolean = b 'Compliant
</pre>

