<p>If no <code>AssemblyVersionAttribute</code> is provided, the same default version will be used for every build. Since the version number is used by
The .NET Framework to uniquely identify an assembly this can lead to broken dependencies.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Imports System.Reflection

&lt;Assembly: AssemblyTitle("MyAssembly")&gt; ' Noncompliant

Namespace MyLibrary

End Namespace
</pre>
<h2>Compliant Solution</h2>
<pre>
Imports System.Reflection

&lt;Assembly: AssemblyTitle("MyAssembly")&gt;
&lt;Assembly: AssemblyVersion("1.1.125.0")&gt;

Namespace MyLibrary

End Namespace
</pre>
<h2>See</h2>
<p><a href="https://docs.microsoft.com/en-us/dotnet/standard/assembly/versioning">Microsoft documentation - Assembly Versioning</a></p>

