<p>Having all branches in a <code>Select Case</code> or <code>If</code> chain with the same implementation is an error. Either a copy-paste error was
made and something different should be executed, or there shouldn't be a <code>Select Case</code> / <code>If</code> chain at all.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Dim b As Integer = If(a &gt; 12, 4, 4)

If b = 0 Then
    DoTheThing()
Else
    DoTheThing()
End If

Select Case i
    Case 1
        DoSomething()
    Case 2
        DoSomething()
    Case 3
        DoSomething()
    Case Else
        DoSomething()
End Select
</pre>
<h2>Exceptions</h2>
<p>This rule does not apply to <code>If</code> chains without <code>Else</code>-s, or to <code>Select Case</code>-es without <code>Case Else</code>
clauses.</p>
<pre>
If b = 0 Then ' No issue, this could have been done on purpose to make the code more readable
    DoTheThing()
ElseIf
    DoTheThing()
End If
</pre>

