<p>The .NET framework class library provides methods for retrieving custom attributes. Sealing the attribute eliminates the search through the
inheritance hierarchy, and can improve performance.</p>
<p>This rule raises an issue when a public type inherits from <code>System.Attribute</code>, is not abstract, and is not sealed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Public Class MyAttribute    ' Noncompliant
    Inherits Attribute

    Public ReadOnly Property Name As String

    Public Sub New(Name As String)
        Me.Name = Name
    End Sub

End Class
</pre>
<h2>Compliant Solution</h2>
<pre>
Public NotInheritable Class MyAttribute
    Inherits Attribute

    Public ReadOnly Property Name As String

    Public Sub New(Name As String)
        Me.Name = Name
    End Sub

End Class
</pre>

