<p>The <code>Obsolete</code> attribute can be applied with or without a message argument. Marking something <code>Obsolete</code> without including
advice on why itâ€™s obsolete or what to use instead will lead maintainers to waste time trying to figure those things out.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class Car
{

  [Obsolete]  // Noncompliant
  public void CrankEngine(int turnsOfCrank)
  { ... }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class Car
{

  [Obsolete("Replaced by the automatic starter")]
  public void CrankEngine(int turnsOfCrank)
  { ... }
}
</pre>

