<p>This rule is meant to be used as a way to track code which is marked as being deprecated. Deprecated code should eventually be removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
[Obsolete] // Noncompliant
void Method()
{
    // ..
}
</pre>

