<h2>Why is this an issue?</h2>
<p>Unused parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.</p>
<p>This rule raises an issue when a <code>private</code> method or constructor of a class/struct takes a parameter without using it.</p>
<h3>Noncompliant code example</h3>
<pre>
private void DoSomething(int a, int b) // "b" is unused
{
    Compute(a);
}

private void DoSomething2(int a) // value of "a" is unused
{
    a = 10;
    Compute(a);
}
</pre>
<h3>Compliant solution</h3>
<pre>
private void DoSomething(int a)
{
    Compute(a);
}

private void DoSomething2()
{
    var a = 10;
    Compute(a);
}
</pre>
<h3>Exceptions</h3>
<p>This rule doesnâ€™t raise any issue in the following contexts:</p>
<ul>
  <li> The <code>this</code> parameter of extension methods. </li>
  <li> Methods decorated with attributes. </li>
  <li> Empty methods. </li>
  <li> Methods which only throw <code>NotImplementedException</code>. </li>
  <li> Main methods. </li>
  <li> <code>virtual</code>, <code>override</code> methods. </li>
  <li> interface implementations. </li>
</ul>

