<h2>Why is this an issue?</h2>
<p>There are several reasons for a method not to have a method body:</p>
<ul>
  <li> It is an unintentional omission, and should be fixed. </li>
  <li> It is not yet, or never will be, supported. In this case a <code>NotSupportedException</code> should be thrown. </li>
  <li> The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override. </li>
</ul>
<h3>Noncompliant code example</h3>
<pre>
public override void DoSomething()
{
}

public override void DoSomethingElse()
{
}
</pre>
<h3>Compliant solution</h3>
<pre>
public override void DoSomething()
{
  // Do nothing because of X and Y.
}

public override void DoSomethingElse()
{
  throw new NotSupportedException();
}
</pre>
<h3>Exceptions</h3>
<p>The following methods are ignored:</p>
<ul>
  <li> empty <code>virtual</code> methods, </li>
  <li> empty methods that override an <code>abstract</code> method, </li>
  <li> empty overrides in test assemblies. </li>
</ul>

