<h2>Why is this an issue?</h2>
<p>Events that are not invoked anywhere are dead code, and thereâ€™s no good reason to keep them in the source.</p>
<h3>Noncompliant code example</h3>
<pre>
class UninvokedEventSample
{
    private event Action&lt;object, EventArgs&gt; Happened; // Noncompliant

    public void RegisterEventHandler(Action&lt;object, EventArgs&gt; handler)
    {
        Happened += handler; // we register some event handlers
    }

    public void RaiseEvent()
    {
        if (Happened != null)
        {
            // Happened(this, null); // the event is never triggered, because this line is commented out.
        }
    }
}
</pre>

