<p><code>NotImplementedException</code> is often used to mark methods which must be implemented for the overall functionality to be complete, but
which the developer wants to implement later. That's as opposed to the <code>NotSupportedException</code> which is thrown by methods which are
required by base classes or interfaces, but which are not appropriate to the current class.</p>
<p>This rule raises an exception when <code>NotImplementedException</code> is thrown.</p>
<h2>Noncompliant Code Example</h2>
<pre>
void doTheThing()
{
    throw new NotImplementedException();
}
</pre>
<h2>Exceptions</h2>
<p>Exceptions derived from <code>NotImplementedException</code> are ignored.</p>

