<h2>Why is this an issue?</h2>
<p>Creating a new <code>Exception</code> without actually throwing it is useless and is probably due to a mistake.</p>
<h3>Noncompliant code example</h3>
<pre>
if (x &lt; 0)
{
  new ArgumentException("x must be nonnegative");
}
</pre>
<h3>Compliant solution</h3>
<pre>
if (x &lt; 0)
{
  throw new ArgumentException("x must be nonnegative");
}
</pre>

