<p>Creating an extension method that extends <code>object</code> is not recommended because it makes the method available on <em>every</em> type.
Extensions should be applied at the most specialized level possible, and that is very unlikely to be <code>object</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public static class MyExtensions
{
    public static void SomeExtension(this object obj) // Noncompliant
    {
        // ...
    }
}
</pre>

