<p>Recursion is acceptable in methods, where you can break out of it. But with class types, you end up with code that will compile but not run if you
try to instantiate the class.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class C1&lt;T&gt;
{
}
class C2&lt;T&gt; : C1&lt;C2&lt;C2&lt;T&gt;&gt;&gt; // Noncompliant
{
}

...
var c2 = new C2&lt;int&gt;();
</pre>

