<p>It is highly suspicious when a value is saved for a key or index and then unconditionally overwritten. Such replacements are likely in error.</p>
<h2>Noncompliant Code Example</h2>
<pre>
towns[x] = "London";
towns[x] = "Chicago";  // Noncompliant
</pre>

