<p>Unused parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.</p>
<p>This rule raises an issue when a <code>private</code> method of a class/struct takes a parameter without using it.</p>
<h2>Noncompliant Code Example</h2>
<pre>
private void DoSomething(int a, int b) // "b" is unused
{
    Compute(a);
}

private void DoSomething2(int a) // value of "a" is unused
{
    a = 10;
    Compute(a);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
private void DoSomething(int a)
{
    Compute(a);
}

private void DoSomething2()
{
    var a = 10;
    Compute(a);
}
</pre>
<h2>Exceptions</h2>
<p>This rule doesn't raise any issue in the following contexts:</p>
<ul>
  <li> The <code>this</code> parameter of extension methods. </li>
  <li> Methods decorated with attributes. </li>
  <li> Empty methods. </li>
  <li> Methods which only throw <code>NotImplementedException</code>. </li>
  <li> Main methods. </li>
  <li> <code>virtual</code>, <code>override</code> methods. </li>
  <li> interface implementations. </li>
</ul>
<h2>See</h2>
<ul>
  <li> <a href="https://wiki.sei.cmu.edu/confluence/x/5dUxBQ">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never executed
  </li>
</ul>

