<p>Consistently using the <code>&amp;</code> operator for string concatenation make the developer intentions clear.</p>
<p><code>&amp;</code>, unlike <code>+</code>, will convert its operands to strings and perform an actual concatenation.</p>
<p><code>+</code> on the other hand can be an addition, or a concatenation, depending on the operand types.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Module Module1
    Sub Main()
        Console.WriteLine("1" + 2) ' Noncompliant - will display "3"
    End Sub
End Module
</pre>
<h2>Compliant Solution</h2>
<pre>
Module Module1
    Sub Main()
        Console.WriteLine(1 &amp; 2)   ' Compliant - will display "12"
        Console.WriteLine(1 + 2)   ' Compliant - but will display "3"
        Console.WriteLine("1" &amp; 2) ' Compliant - will display "12"
    End Sub
End Module
</pre>

