<p>The best way to determine the type of a generic method is by inference based on the type of argument that is passed to the method. This is not
possible when a parameter type is missing from the argument list.</p>
<h2>Noncompliant Code Example</h2>
<pre>
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void MyMethod&lt;T&gt;()  // Noncompliant
    {
    }
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void MyMethod&lt;T&gt;(T param)
    {
    }
  }
}
</pre>

