<p>Caller information attributes (<code>CallerFilePathAttribute</code>, <code>CallerLineNumberAttribute</code>, and
<code>CallerMemberNameAttribute</code>) provide a way to get information about the caller of a method through optional parameters. But the arguments
for these optional parameters are only generated if they are not explicitly defined in the call. Thus, specifying the argument values defeats the
purpose of the attributes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
void TraceMessage(string message,
  [CallerMemberName] string memberName = "",
  [CallerFilePath] string filePath = "",
  [CallerLineNumber] int lineNumber = 0)
{
  /* ... */
}

void MyMethod()
{
  TraceMessage("my message", "MyMethod"); // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
void TraceMessage(string message,
  [CallerMemberName] string memberName = "",
  [CallerFilePath] string filePath = "",
  [CallerLineNumber] int lineNumber = 0)
{
  /* ... */
}

void MyMethod()
{
  TraceMessage("my message");
}
</pre>

