<p>There's no point in checking a variable against the value you're about to assign it. Save the cycles and lines of code, and simply perform the
assignment.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (x != a)  // Noncompliant; why bother?
{
  x = a;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
x = a;
</pre>

