<p>Numbers can be shifted with the <code>&lt;&lt;</code> and <code>&gt;&gt;</code> operators, but the right operand of the operation needs to be an
<code>int</code> or a type that has an implicit conversion to <code>int</code>. However, with <code>dynamic</code>, the compiler's type checking is
turned off, so you can pass anything to a shift operator and have it compile. And if the argument can't be converted to <code>int</code> at runtime,
then a <code>RuntimeBinderException</code> will be raised.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Dim o As Object = 5
Dim x As Integer = 5

x = o &gt;&gt; 5 ' Noncompliant
x = x &gt;&gt; o ' Noncompliant
</pre>
<h2>Exceptions</h2>
<p>This rule does not raise when the left or the right expression is <code>Nothing</code>.</p>
<pre>
x = Nothing &gt;&gt; 5
x = 5 &gt;&gt; Nothing
</pre>

