<p>Numbers are infinite, but the types that hold them are not. Each numeric type has hard upper and lower bounds. Try to calculate or assign numbers
beyond those bounds, and the result will be a value that has silently wrapped around from the expected positive value to a negative one, or vice
versa.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public int getTheNumber(int positiveInt) {
  int num = Integer.MAX_VALUE;
  return num + positiveInt;  // Noncompliant
}
</pre>

