<p><code>switch</code> statements are useful when there are many different cases depending on the value of the same expression.</p>
<p>For just one or two cases however, the code will be more readable with <code>if</code> statements.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Select Case variable
    Case 0
        doSomething()
    Case Else
        doSomethingElse()
End Select
</pre>
<h2>Compliant Solution</h2>
<pre>
If variable = 0 Then
    doSomething()
Else
    doSomethingElse()
End If
</pre>
<h2>See</h2>
<ul>
  <li> MISRA C:2012, 16.6 - Every switch statement shall have at least two switch-clauses </li>
</ul>

