<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate.</p>
<p>This rule allows to check that all parameter names match a provided regular expression.</p>
<p>The default configuration is the one recommended by Microsoft:</p>
<ul>
  <li> Camel casing, starting with a lower case character, e.g. backColor </li>
  <li> Short abbreviations of 2 letters can be capitalized only when not at the beginning, e.g. id, productID </li>
  <li> Longer abbreviations need to be lower cased, e.g. html </li>
</ul>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression <code>^[a-z][a-z0-9]*([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</code></p>
<pre>
Module Module1
    Sub GetSomething(ByVal ID As Integer) ' Noncompliant
    End Sub
End Module
</pre>
<h2>Compliant Solution</h2>
<pre>
Module Module1
    Sub GetSomething(ByVal id As Integer) ' Compliant
    End Sub
End Module
</pre>

