<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all namespace names match a provided regular
expression.</p>
<p>The default configuration is the one recommended by Microsoft:</p>
<ul>
  <li> Pascal casing, starting with an upper case character, e.g. Microsoft, System </li>
  <li> Short abbreviations of 2 letters can be capitalized, e.g. System.IO </li>
  <li> Longer abbreviations need to be lower cased </li>
</ul>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression: <code>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?(\.([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2}))*$</code></p>
<pre>
Namespace foo  ' Noncompliant
End Namespace
</pre>
<h2>Compliant Solution</h2>
<pre>
Namespace Foo  ' Compliant
End Namespace
</pre>

