<p>Fields should not be part of an API, and therefore should always be private. Indeed, they cannot be added to an interface for instance, and
validation cannot be added later on without breaking backward compatibility. Instead, developers should encapsulate their fields into properties.
Explicit property getters and setters can be introduced for validation purposes or to smooth the transition to a newer system.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Class Foo
    Public Foo = 42          ' Noncompliant
End Class
</pre>
<h2>Compliant Solution</h2>
<pre>
Class Foo
    Public Property Foo = 42 ' Compliant
End Class
</pre>
<h2>Exceptions</h2>
<p><code>Shared</code> and <code>Const</code> fields are ignored.</p>

